include/chemfiles/files/TRRFile.hpp

```diff
@@ -0,0 +1,51 @@
+// Chemfiles, a modern library for chemistry file reading and writing
+// Copyright (C) Guillaume Fraux and contributors -- BSD license
+
+#ifndef CHEMFILES_TRR_FILE_HPP
+#define CHEMFILES_TRR_FILE_HPP
+
+#include "chemfiles/File.hpp"
+#include "xdrfile.h"
+#include "xdrfile_trr.h"
+
+namespace chemfiles {
+
+/// Simple RAII capsule for `XDRFILE*`, handling the creation and
+/// destruction of the file as needed.
+/// Reads the file header and stores the offsets for individual frames.
+class TRRFile final : public File {
+  public:
+    TRRFile(std::string path, File::Mode mode);
+    ~TRRFile() override;
+    TRRFile(TRRFile&&) noexcept = default;
+    TRRFile& operator=(TRRFile&&) = default;
+    TRRFile(TRRFile const&) = delete;
+    TRRFile& operator=(TRRFile const&) = delete;
+
+    /// get the number of frames/steps in the file, as indicated in the file header
+    unsigned long nframes() const;
+    /// get the offset corresponding to a specific frame/step
+    int64_t offset(int64_t step) const;
```

Comment:
Shoul the step be a `size_t` value rather than a signed integer?

Reply:
The same apply to XTCFile =)

Reply:
Done, also for XTC.